code/_onclick/click.dm

```diff
@@ -238,16 +238,13 @@
 	// A is a turf or is on a turf, or in something on a turf (pen in a box); but not something in something on a turf (pen in a box in a backpack)
 	sdepth = A.storage_depth_turf()
 	if (isturf(A) || isturf(A.loc) || (sdepth != -1 && sdepth <= 1))
-		if (A.Adjacent(src) || (W && W == get_active_hand() && (istype(W, /obj/item/weapon/barrier))) && A.rangedAdjacent(src)) // see adjacent.dm
```

Comment:
Why?

Reply:
because there is an istype check that says we cannot click UNLESS we are adjacent OR we are a barrier and can use it at range

Reply:
there is literally a hardcoded check that gives barriers infinite range usage here